pkg/secrets/push_to_file/secrets_groups.go

```diff
@@ -0,0 +1,235 @@
+
+			secretsPathPrefix, err := parseSecretsPathPrefix(groupName, annotations[SECRETS_GROUP_POLICY_PATH_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			secrets, err := parseSecretsPaths(groupName, annotations[SECRETS_GROUP_PREFIX+groupName], secretsPathPrefix)
+			if err != nil {
+				return nil, err
+			}
+
+			filePath, err := parseFilePath(groupName, annotations[SECRETS_GROUP_FILE_PATH_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			fileType, err := parseFileType(groupName, annotations[SECRETS_GROUP_FILE_TYPE_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			fileTemplate, err := parseFileTemplate(groupName, annotations[SECRETS_GROUP_FILE_TEMPLATE_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			if fileTemplate != nil {
+				fileType = FILE_TYPE_NONE
+			}
+
+			groupInfo := SecretsGroupInfo{
+				Secrets:      secrets,
+				FilePath:     filePath,
+				FileType:     fileType,
+				FilePerms:    DEFAULT_FILE_PERMS,
+				FileTemplate: fileTemplate,
+			}
+
+			err = validateGroupInfo(groupName, groupInfo)
+			if err != nil {
+				return nil, err
+			}
+
+			/*
+				if groupInfo.FileTemplate != nil {
+					groupInfo.FileTemplate.Execute(os.Stdout, groupInfo.Secrets)
+				}
+			*/
+
+			secretsGroups[groupName] = groupInfo
+		}
+	}
+
+	return secretsGroups, nil
+}
+
+func parseSecretsPaths(groupName string, secretsPaths string, secretsPathsPrefix string) (SecretsPaths, error) {
+	secrets := make(SecretsPaths)
+
+	unpacked := []interface{}{}
+	if err := yaml.Unmarshal([]byte(secretsPaths), &unpacked); err != nil {
+		return nil, log.RecordedError(messages.CSPFK051E, err.Error(), groupName)
+	}
+
+	insertSecret := func(name string, secretPath string) error {
+		secretPath = path.Clean(secretPath)
+		if secretPath == "." {
+			return log.RecordedError(messages.CSPFK051E, "Invalid secret path", groupName)
+		}
+		secrets[name] = secretsPathsPrefix + secretPath
+
+		return nil
+	}
+
+	for _, secret := range unpacked {
+		switch val := secret.(type) {
+		case string:
+			{
+				name := val[strings.LastIndex(val, "/")+1:]
+				if name == "" {
+					return nil, log.RecordedError(messages.CSPFK051E, "Invalid secret name", groupName)
+				}
+
+				if err := insertSecret(name, val); err != nil {
+					return nil, err
+				}
+			}
+
+		case map[string]interface{}:
+			{
+				for alias, secretPath := range val {
+					secretPath, _ := secretPath.(string)
+					if err := insertSecret(alias, secretPath); err != nil {
+						return nil, err
+					}
+				}
+			}
+
+		default:
+			return nil, log.RecordedError(messages.CSPFK051E, "Unknown secret format", groupName)
+		}
+	}
+
+	if len(secrets) == 0 {
+		return nil, log.RecordedError(messages.CSPFK051E, "No valid secrets found", groupName)
+	}
+
+	return secrets, nil
+}
+
+func parseSecretsPathPrefix(groupName string, secretsPathPrefix string) (string, error) {
```

Comment:
A comment describing what's being done here would help, e.g. it defaults to the root policy path `/`, otherwise adds a trailing `/` if necessary, or something like that.

Reply:
Done